### PR TITLE
Add missing SC on skills properties

### LINKED ISSUE #54: Some missing SC on skills properties
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/DerpyProjectGroup/topaz/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/DerpyProjectGroup/topaz/blob/info/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: freedom

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
As was recently mentioned in the Discord channel, some skillchain properties appear to be missing from certain skills. 

The linked file may be useful (from the skillchains addon). 
https://github.com/Ivaar/Ashita-addons/blob/master/Skillchains/skills.lua

### PR DESCRIPTION
**Pull request by illzbee**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

"Closes #54"
Adds missing SC on skills properties
Used https://github.com/Ivaar/Skillchains/blob/master/skills.lua for reference and ffxiclopedia to double check on removal/addition to "main/sub skillchains"

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
Changelist is huge for BLU spells, which ones added a tertiary SC property?

**Comment by WinterSolstice8** [comment]
WS properties check out

**Comment by zach2good** [comment]
For the BLU changes, are you going to hook up reading the new SQL column in core?

https://github.com/LandSandBoat/server/blob/d9d3a7b6494d17f2178b140c9b0f6d9e983d7459/src/map/spell.cpp#L541

**Comment by illzbee** [comment]
> Changelist is huge for BLU spells, which ones added a tertiary SC property?

Sorry should have mentioned only one added a tertiary sc "Quadrastrike" whole list was removing ' ' maybe should've pushed the changes first then removed ' ' after...

**Comment by illzbee** [comment]
> For the BLU changes, are you going to hook up reading the new SQL column in core?
> 
> https://github.com/LandSandBoat/server/blob/d9d3a7b6494d17f2178b140c9b0f6d9e983d7459/src/map/spell.cpp#L541

Should be good to go now. Thanks Zach!

Or not... total fail on building. -_-

I am at a loss, if I change to this 

        const char* blueQuery = "SELECT blue_spell_list.spellid, blue_spell_list.mob_skill_id, blue_spell_list.set_points, \
                                blue_spell_list.trait_category, blue_spell_list.trait_category_weight, blue_spell_list.primary_sc, \
                                blue_spell_list.secondary_sc, blue_spell_list.tertiary_sc, spell_list.content_tag \
                             FROM blue_spell_list JOIN spell_list on blue_spell_list.spellid = spell_list.spellid;";

        ret = sql->Query(blueQuery);

        if (ret != SQL_ERROR && sql->NumRows() != 0)
        {
            while (sql->NextRow() == SQL_SUCCESS)
            {
                char* contentTag = (char*)sql->GetData(8);
                if (!luautils::IsContentEnabled(contentTag))
                {
                    continue;
                }

                // Sanity check the spell ID
                uint16 spellId = sql->GetIntData(0);

                if (PSpellList[spellId] == nullptr)
                {
                    ShowWarning("spell::LoadSpellList Tried to load nullptr blue spell (%u)", spellId);
                    continue;
                }

                ((CBlueSpell*)PSpellList[spellId])->setMonsterSkillId(sql->GetIntData(1));
                ((CBlueSpell*)PSpellList[spellId])->setSetPoints(sql->GetIntData(2));
                ((CBlueSpell*)PSpellList[spellId])->setTraitCategory(sql->GetIntData(3));
                ((CBlueSpell*)PSpellList[spellId])->setTraitWeight(sql->GetIntData(4));
                ((CBlueSpell*)PSpellList[spellId])->setPrimarySkillchain(sql->GetIntData(5));
                ((CBlueSpell*)PSpellList[spellId])->setSecondarySkillchain(sql->GetIntData(6));
                ((CBlueSpell*)PSpellList[spellId])->setTertiarySkillchain(sql->GetIntData(7));
                PMobSkillToBlueSpell.insert(std::make_pair(sql->GetIntData(1), spellId));
            }
        }

build fails

removing
                ((CBlueSpell*)PSpellList[spellId])->setTertiarySkillchain(sql->GetIntData(7));
build completes.

Just making a big mess with commits now...

**Comment by WinterSolstice8** [comment]
`no member named 'setTertiarySkillchain' in 'CBlueSpell';`
that means all the plumbing for tertiary SC effect is missing.  you'd have to add all the calls to get/set that during skillchains to make it properly work. If you'd like, you could PR just the SQL changes and we can handle the C++ part. You could use this as a learning experience as well.

**Comment by zach2good** [comment]
You've added your declaration for your new methods in `blue_spell.h`, but you haven't added any definitions in `blue_spell.cpp`

https://stackoverflow.com/questions/1410563/what-is-the-difference-between-a-definition-and-a-declaration

Like Primary has these in the header file:
```cpp
    uint8  getPrimarySkillchain() const;
    void   setPrimarySkillchain(uint8 sc);
```

and these in the cpp file:
```cpp

uint8 CBlueSpell::getPrimarySkillchain() const
{
    return m_PrimarySkillchain;
}

void CBlueSpell::setPrimarySkillchain(uint8 sc)
{
    m_PrimarySkillchain = sc;
}
```

**Comment by illzbee** [comment]
> You've added your declaration for your new methods in `blue_spell.h`, but you haven't added any definitions in `blue_spell.cpp`
> 
> https://stackoverflow.com/questions/1410563/what-is-the-difference-between-a-definition-and-a-declaration
> 
> Like Primary has these in the header file:
> 
> ```c++
>     uint8  getPrimarySkillchain() const;
>     void   setPrimarySkillchain(uint8 sc);
> ```
> 
> and these in the cpp file:
> 
> ```c++
> uint8 CBlueSpell::getPrimarySkillchain() const
> {
>     return m_PrimarySkillchain;
> }
> 
> void CBlueSpell::setPrimarySkillchain(uint8 sc)
> {
>     m_PrimarySkillchain = sc;
> }
> ```

yeah I just did this ran a search on all src files for primary to find where I was missing ...lol... thanks again.

File: src/map/blue_spell.cpp [hunk 1]
```diff
@@ -103,6 +103,16 @@ void CBlueSpell::setSecondarySkillchain(uint8 sc)
     m_SecondarySkillchain = sc;
 }
 
+uint8 CBlueSpell::getSecondarySkillchain() const
+{
+    return m_TertiarySkillchain;
+}
+
+void CBlueSpell::setSecondarySkillchain(uint8 sc)
```
Comment (zach2good) [inline src/map/blue_spell.cpp]: Tertiary *

Comment (illzbee) [reply src/map/blue_spell.cpp]: Did I spell something incorrect?

Comment (illzbee) [reply src/map/blue_spell.cpp]: oh duh lol dont mind me

Comment (illzbee) [reply src/map/blue_spell.cpp]: rushing before work isn't always good

**Review by zach2good — APPROVED** [review]

**Pull request merged**